spec/java/util/LinkedList.lsl

```diff
@@ -664,3 +669,58 @@ automaton ListItr: int(
     }
     
 }
+
+
+
+@Private
+@Implements('java.util.Iterator')
+@WrapperMeta(
+    src='java.util.ListItr',
+    dst='org.utbot.engine.overrides.collections.UtDescendingIterator',
+    matchInterfaces=true,
+)
+automaton DescendingIterator: int(
+	//Do we can write in such way or not ??
+	var itr = new ListItr(state=Created,
```

Comment:
Automata variables cannot have references to other automata.

Reply:
@dog-m But how i must work in such situation ? I must dublicate hasPrevious(), previous() and remove() realization ? 

```
private class DescendingIterator implements Iterator<E> {
        private final ListItr itr = new ListItr(size());
        public boolean hasNext() {
            return itr.hasPrevious();
        }
        public E next() {
            return itr.previous();
        }
        public void remove() {
            itr.remove();
        }
    }
```

Reply:
If it is unavoidable, yes, code duplication is an acceptable solution while we cannot have automata inheritance or templates.